src/Module.Server/Common/BreakableWeaponsBehaviorClient.cs

```diff
@@ -0,0 +1,68 @@
+﻿using Crpg.Module.Common.Network;
+using Crpg.Module.Notifications;
+using NetworkMessages.FromServer;
+using TaleWorlds.Core;
+using TaleWorlds.Engine;
+using TaleWorlds.Library;
+using TaleWorlds.MountAndBlade;
+using TaleWorlds.MountAndBlade.Diamond;
+using TaleWorlds.PlayerServices;
+using Timer = TaleWorlds.Core.Timer;
+
+namespace Crpg.Module.Common;
+
+internal class BreakableWeaponsBehaviorClient : MissionNetwork
+{
+    public int LastRoll { get; private set; }
+    public int LastBlow { get; private set; }
+
+    public override void OnAgentBuild(Agent agent, Banner banner)
+    {
+        if (agent == null)
+        {
+            return;
+        }
+
+        if (agent.Equipment == null)
+        {
+            return;
+        }
+
+        for (int i = 0; i < 5; i++)
```

Comment:
I would do it like that:
`for (var i = EquipmentIndex.Weapon0; i <= EquipmentIndex.ExtraWeaponSlot; i += 1)`

It's like this in CrpgSpawningBehaviorBase aswell and is more robust and clear.